### PR TITLE
feat(trace): UI lazy loading of spans

### LINKED ISSUE #3917: Trace lazy loading [ENHANCEMENT]
**Is your feature request related to a problem? Please describe.**
My traces are large and it's taking > 1 minutes to load the project page on the web UI

**Describe the solution you'd like**
Is it possible to make phoenix web UI lazily load traces' children spans? Or is it possible to limit the pages size on the UI?

**Describe alternatives you've considered**
N/A

**Additional context**
https://arize-ai.slack.com/archives/C04R3GXC8HK/p1721163004296329

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3917

Makes it so that the span lazily loads only on selection.

### CODE REVIEW

File: app/src/pages/trace/SpanDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,1667 @@
...
+    {
+      spanId: spanNodeId,
+    }
+  );
+
+  if (span.__typename !== "Span") {
+    throw new Error(
+      "Expected a span, but got a different type" + span.__typename
+    );
+  }
```
Comment (Parker-Stafford) [inline app/src/pages/trace/SpanDetails.tsx:213]: is this basically typeguarding or an actual error we would expect? seems impossible without a major bug

File: app/src/pages/trace/SpanDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,1667 @@
...
+
+  const hasExceptions = useMemo<boolean>(() => {
+    return spanHasException(span);
+  }, [span]);
+  const showAnnotations = useFeatureFlag("annotations");
+  return (
+    <Flex direction="column" flex="1 1 auto" height="100%">
```
Comment (Parker-Stafford) [inline app/src/pages/trace/SpanDetails.tsx:220]: stuff below here is just copied over from old trace page span details correct?

Comment (mikeldking) [reply app/src/pages/trace/SpanDetails.tsx:220]: yes

File: app/src/pages/trace/SpanDetails.tsx (hunk 1 above)
Comment (mikeldking) [reply app/src/pages/trace/SpanDetails.tsx:213]: yup, it's a typequard for the node interface

**Review by Parker-Stafford — APPROVED** [review]

**Pull request merged**